sirenabailie/Fan-Fusion-FE
# Check user/ym

## Detailed Description


- Created a checkUser endpoint in the auth.js (this will be used when a user logs into our application)
- What it does is pull the logged-in users uid and see if there's a user in our database that has a matching uid. If it does then the user object in our database will get saved to our useAuth. 
- Updated the AuthConext.js file so it has the functionality we need in order to save the user object in our useAuth hook. 
- Updataed the client.js so it only has the relevant information we need. 
- Lastly, I added a console.log in the main page.js so we can see what it happening when a user logs in. 

## Related Issues

## How to Test


1.  Go to pgAdmin. Open all rows for the Users data. Select one of the user's and change the uid to your uid. When you change it you need to click the save data changes icon. We need to do this in order to connect you to a user that we currently have. Feel free to change the name of the user as well if you wish. 
2. Pull down branch and first update your env file. It should look like this.  When we approve this PR you need to ensure your env still looks like this. 
```
NEXT_PUBLIC_FIREBASE_API_KEY="AIzaSyAewtzVLVJl0Cc-ysODgf92CsBB4m1Qp1c"
NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN="fan-fusion.firebaseapp.com"

NEXT_PUBLIC_DATABASE_URL= https://localhost:7238

```
3. Make sure that you are running swagger in the back before running npm run dev. 
4. Make sure you are logged out and when you log in check your console to see if it is logging our the fbUser as well as the user object you selected in the BE database. 
## Motivation and Context

The reason that we need this is because we need to store the userObj so that when we need to grab the userId for some of our endpoints then we can easily grab it by using useAuth. 

## Screenshots (if applicable)

## Types of Changes

- [x ] ✨ New feature

## 💬 Additional Notes
